evm/logic/system.py

```diff
@@ -73,7 +81,10 @@ def selfdestruct_eip161(computation):
 
 
 def _selfdestruct(computation, beneficiary):
-    with computation.vm.state_db() as state_db:
+    with computation.vm.state_db(
+        read_list=computation.msg.read_list,
+        write_list=computation.msg.write_list
```

Comment:
This repetition makes me think that we should add a convenience method `Computation.get_state_db()` which auto-populates these values based on the current message.

Reply:
Good idea, wasn't happy with this either. My first instinct was to add the defaults to `VM.state_db()` but this obviously doesn't work because `VM` doesn't know about the current tx/msg.

Naming convention wise, should it be `get_state_db` or `state_db`? Looks like the difference right now is that `VM.state_db` is a context manager whereas `ChainDB.get_state_db` is not. So `Computation.state_db` seems to be more consistent.